libsolidity/interface/SMTSolverCommand.cpp

```diff
@@ -0,0 +1,82 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+#include <libsolidity/interface/SMTSolverCommand.h>
+
+#include <liblangutil/Exceptions.h>
+
+#include <libsolutil/CommonIO.h>
+#include <libsolutil/Exceptions.h>
+
+#include <boost/algorithm/string/join.hpp>
+#include <boost/algorithm/string/predicate.hpp>
+#include <boost/filesystem.hpp>
+#include <boost/filesystem/fstream.hpp>
+#include <boost/process.hpp>
+
+using solidity::langutil::InternalCompilerError;
+using solidity::util::errinfo_comment;
+
+using namespace std;
+
+namespace solidity::frontend
+{
+
+SMTSolverCommand::SMTSolverCommand(string _solverCmd) : m_solverCmd(_solverCmd) {}
+
+ReadCallback::Result SMTSolverCommand::solve(string const& _kind, string const& _query)
+{
+	try
+	{
+		if (_kind != ReadCallback::kindString(ReadCallback::Kind::SMTQuery))
+			solAssert(false, "SMTQuery callback used as callback kind " + _kind);
+
+		auto tempDir = boost::filesystem::temp_directory_path();
+		auto queryFileName = tempDir / "query.smt2";
```

Comment:
I thought you were creating a new dir but after running it, I see that you just leave `query.smt2` directly in the tmp dir. That's not too great, especially when someone tries to run multiple instances of `solc` in parallel. You should create the file/dir with a unique name, and preferably delete it once you no longer need it.

I actually have a utility for handling that better, and I'd suggest using it, though you'd have to move it from tests to solutil:
https://github.com/ethereum/solidity/blob/9db2da0385c5abec0d1c3eab468648ec85fb5caa/test/TemporaryDirectory.h#L41-L56

Reply:
Yea good point. Fixed it.